OpenTelemetry ingestion sampling guide (#16150)
DataDog/documentation: content/en/tracing/guide/ingestion_sampling_with_opentelemetry.md

### SDK-level sampling

At the SDK level, you can implement _head-based sampling_, which is when the sampling decision is made at the beginning of the trace. This type of sampling is particularly useful for high throughput applications, for which you know that you do not need visibility over 100% of the traffic to monitor the application health. It can also help control the overhead introduced by OpenTelemetry.

[TraceIdRatioBased][5] and [ParentBased][6] are the SDK's built-in samplers that allow you to implement deterministic head-based sampling based on the `trace_id` at the SDK level.

> With head-based sampling, the APM metrics are computed **on the reduced traffic**, since only the sampled traffic is sent to the OpenTelemetry Collector or Datadog Agent, which is where the metrics calculation is done.

Author: ```suggestion
With head-based sampling, the APM metrics are computed **on the sampled traffic**, since only the sampled traffic is sent to the OpenTelemetry Collector or Datadog Agent, which is where the metrics calculation is done.
```

> To get accurate stats, you can upscale the metrics by using [formulas and functions][7] in Datadog dashboards and monitors, provided you know the configured sampling rate in the SDK.

Author: ```suggestion
To get accurate stats, you can upscale the metrics by using [formulas and functions][7] in Datadog dashboards and monitors, provided that you know the configured sampling rate in the SDK.
```

Use the [ingestion volume control guide][8] to read more about the implications of setting up trace sampling on trace analytics monitors and metrics from spans.

### Collector-level sampling

At the OpenTelemetry collector level, you can do _tail-based sampling_, which allows you to define more advanced rules to keep accrued visibility over error or high latency traces.

The [Tail Sampling Processor][9] and [Probabilistic Sampling Processor][10] allow you to sample traces based on a set of rules at the collector level.

**Note**: Tail sampling's main limitation is that all spans for a given trace must be received by the same collector instance for effective sampling decisions. If the trace is distributed across multiple collector instances, there’s a risk that some parts of a trace are dropped whereas some other parts of the same trace are sent to Datadog.

> With collector-level tail-based sampling, the APM metrics are computed on the all traffic that arrives at the collector. Use the [ingestion volume control guide][11] to read more about the implications of setting up trace sampling on trace analytics monitors and metrics from spans.

Author: ```suggestion
With collector-level tail-based sampling, the APM metrics are computed on the sampled traffic that the Datadog Exporter sees. Use the [ingestion volume control guide][11] to read more about the implications of setting up trace sampling on trace analytics monitors and metrics from spans.
```
